clients/spark/core/src/main/scala/io/treeverse/clients/Exporter.scala

```diff
@@ -42,15 +43,18 @@ def exportFrom(branch: String, prevCommitID: String): Unit = {
 
     // pin Exporter field to avoid serialization
     val dst = dstRoot
-    val actionsDF = spark.sql("SELECT 'copy' as action, new_commit.* FROM new_commit " +
-      "LEFT JOIN prev_commit " +
-      "ON (new_commit.etag = prev_commit.etag AND new_commit.key = prev_commit.key) " +
-      "WHERE prev_commit.key is NULL " +
-      "UNION " +
-      "SELECT 'delete' as action, prev_commit.* FROM prev_commit " +
-      "LEFT OUTER JOIN new_commit " +
-      "ON (new_commit.key = prev_commit.key) " +
-      "WHERE new_commit.key is NULL")
+
+    val actionsDF = spark.sql("WITH raw_rows AS (SELECT n.key as nkey, n.address as naddress, n.etag as netag, " +
```

Comment:
I think I would find this easier to read as a subselect than as a CTE -- in particular the fact that `raw_rows` is elided in the query confuses me.

I.e. it could like something like

```sql
SELECT CASE ... END as action, CASE ... AS key, ... FROM (
    SELECT n.key AS nkey, ... FROM new_commit n FULL OUTER JOIN prev_commit p ...
)
```